.circleci/config.yml

```
@@ -160,524 +220,261 @@ jobs:
-
-    working_directory: ~/repo
-
-    steps:
-      - checkout
-
-      - run:
-          name: install doxygen
-          command: sudo apt-get install doxygen
-
-      - run:
-          name: install dependencies
-          command: |
-            python -m venv env
-            . env/bin/activate
-            pip install -r requirements.txt
-            pip install -r docs/requirements.txt
-
-      - run: *unix-build-ext-template
-
-      - run:
-          name: build docs
-          command: |
-            . env/bin/activate
-            make -C docs/ cpp html
-
-      - store_artifacts:
-          path: ./docs/build/html
-
-      - run:
-          name: doctest
-          command: |
-            . env/bin/activate
-            make -C docs/ doctest
-
-      - run:
-          name: linkcheck
-          command: |
-            . env/bin/activate
-            make -C docs/ linkcheck
-
-  test-linux-cpp11:
-    docker:
-      # just use a python image, all we really want is debian
-      - image: circleci/python:3.6-jessie
-
-    working_directory: ~/repo
-
-    steps:
-      - checkout
-      - run:
-          name: run cpp tests
-          command: |
-            make -C testscpp/ --always-make
-
 workflows:
   tests:
     jobs:
-      - build-manylinux:
-          name: build-<< matrix.manylinux-tag >>_<< matrix.manylinux-arch >>-py<< matrix.python-version >>
+      - build-linux: &build
           matrix:
             parameters:
-              python-version: &python-versions [3.6.8, 3.7.9, 3.8.9, 3.9.4]
-              manylinux-tag: &manylinux-tags ["manylinux1", "manylinux2014"]
-              manylinux-arch: &manylinux-archs ["x86_64"]
+              python-version: &python-versions [3.6.8, 3.7.9, 3.8.9, 3.9.4, 3.10.0]
+              # python-version: &python-versions [3.9.4]
+      - build-linux-aarch64: *build
+      - build-sdist
+      - build-osx: &build-osx
+          matrix:
+            parameters:
+              python-version: *python-versions
+              cibw-arch: [x86_64, universal2]
+            exclude:
+              - python-version: 3.6.8
+                cibw-arch: universal2
+              - python-version: 3.7.9
+                cibw-arch: universal2
+      - build-windows: *build
+      - test-airspeed-velocity
+      - test-doctest
       - test-linux:
           name: test-linux-numpy<< matrix.numpy-version >>-py<< matrix.python-version >>
-          # We could break each of these out into seperate jobs so that we
-          # don't need to wait for the entire above matrix to complete, but
-          # this is cleaner
           requires:
-            - build-manylinux
+            - build-linux
           matrix:
             parameters:
               # test the lowest numpy version and the highest minor of each
               # currently deployed
-              numpy-version: [==1.17.3, <1.19, <1.20, <1.21]
+              numpy-version: [==1.17.3, <1.19, <1.20, <1.21, <1.22]
```

I find this much cleaner, with the same effect:
```suggestion
              numpy-version: [==1.17.3, ~=1.18.0, ~=1.19.0, ~=1.20.0, ~=1.21.0]
```
Also, no need to upgrade in install command then.